# Varpipe_wgs

**Dependencies**

Bcftools 1.11

BEDTools 2.17

BWA 0.7.17

GATK 4.1.6.0

Piccard Tools 2.23.8

Python 2.7.5

Java 1.8.0_131

Samtools 1.11

snpEff 4.3

Trimmomatic 0.39

**Install Instructions**

1. Pull or clone the varpipe_wgs repo 
2. Edit the /tools/clockwork-0.11.3/clockwork script with the right local filepath
3. Add the path in 2 above to your shell path before running the pipeline


**Run Command**

 in the data folder (after adding all input compressed fastq files)

 ./runVarpipeline.sh


**Disclaimer** 

The Laboratory Branch (LB) of the Division of Tuberculosis Elimination developed this bioinformatic pipeline for analyzing whole genome sequencing data generated on Illumina platforms. This is not a controlled document. The performance characteristics as generated at Centers for Disease Control and Prevention (CDC) are specific to the version as written. These documents are provided by LB solely as an example for how this test performed within LB.  The recipient testing laboratory is responsible for generating validation or verification data as applicable to establish performance characteristics as required by the testing laboratoryâ€™s policies, applicable regulations, and quality system standards. These data are only for the sample and specimen types and conditions described in this procedure. Tests or protocols may include hazardous reagents or biological agents. No indemnification for any loss, claim, damage, or liability is provided for the party receiving an assay or protocol. Use of trade names and commercial sources are for identification only and do not constitute endorsement by the Public Health Service, the United States Department of Health and Human Services, or the Centers for Disease Control and Prevention

# TB_varpipe
# TB_varpipe
